### DISCUSSION TITLE
Heap Space when compiling (or configuring Fabric Loom)

### DISCUSSION THREAD

**Discussion by d0x7** [maruohon/minihud#262] [General]
Hey, I wanna add a feature to MiniHud ("private", not interessting for general public) but I fail at compiling it at all even before I did any changes. I'm not really used to gradle because I use maven, but I tried to build it using ```gradle build``` which seems to be stuck at "Configuring project: Fabric Loom" aaand after a few minutes (of no output, hence the "stuck", but I don't think it's stuck) I get a OutOfMemory Java Heap space error. Not the first heap space I ever got, but the first I can't seem to manage.

After googling, you should be able to add ```org.gradle.jvmargs```to build.properties to increase the heap (using -Xmx4 or 8G). But I'm not sure if it just does nothing at all or whether even 8 frickn gigabytes isn't enough to configure loom? I mean, it's not even compiling minihud, it's always just about configuring loom. 

Any ideas or help?
Thanks :)

**Comment by maruohon** [comment]
It won't work in my `build.properties` file, it's parsed "differently" and later. You need it in a file called `gradle.properties` either in the mod directory or in your main/home Gradle directory at `<home directory>/.gradle/gradle.properties`. And I believe 3 GB (or even 2?) should be enough. Although looks like I have mine set to 4.
